modules/gitops-configuring-common-cluster-roles-by-specifying-user-defined-cluster-roles-for-namespace-scoped-instances.adoc

```diff
@@ -0,0 +1,55 @@
+// Module included in the following assemblies:
+//
+// * argocd_instance/setting-up-argocd-instance.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="gitops-configuring-common-cluster-roles-by-specifying-user-defined-cluster-roles-for-namespace-scoped-instances_{context}"]
+= Configuring common cluster roles by specifying user-defined cluster roles for namespace-scoped instances
+
+As a cluster administrator, when you give an Argo CD access to a namespace by using the `argocd.argoproj.io/managed-by` label, the Argo CD assumes `namespace-admin` privileges. The {gitops-title} Operator then automatically creates role bindings for all managed namespaces of the following {gitops-shortname} control plane components:
+
+* Argo CD Application Controller
+* Argo CD server
+* Argo CD ApplicationSet Controller
+
+When you provide namespaces to non-administrator users, for example, development teams, they can use the `namespace-admin` privileges to modify objects such as network policies. Installing an Argo CD instance in these namespaces gives the development teams `admin` privileges and indirectly elevates their assigned privileges. These roles are highly privileged and can delete all resources. As a preventive action, you can define a specific set of reduced permissions to meet your security requirements by configuring common cluster roles for all managed namespaces in the role bindings that the Operator creates for the Argo CD Application Controller and Argo CD server components.
+
+To configure common cluster roles for all managed namespaces, you can specify user-defined cluster roles for the `CONTROLLER_CLUSTER_ROLE` and `SERVER_CLUSTER_ROLE` environment variables in the Operator's `Subscription` object YAML file. As a result, instead of creating the default `admin` role, the Operator uses the existing user-defined cluster roles and creates role bindings for all managed namespaces.
+
+.Prerequisites
+
+* You have logged in to the {OCP} cluster as an administrator.
+
+* You have installed the {gitops-title} Operator on your {OCP} cluster.
+
+.Procedure
+
+. In the *Administrator* perspective, navigate to *Administration* -> *CustomResourceDefinitions*.
+. Find the *Subscription* CRD and click to open it.
+. Select the *Instances* tab and click the *openshift-gitops-operator* subscription.
+. Select the *YAML* tab and make your customization:
+** Specify the user-defined cluster roles for the `CONTROLLER_CLUSTER_ROLE` and `SERVER_CLUSTER_ROLE` environment variables:
++
+.Example Subscription
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: openshift-gitops-operator
+  namespace: openshift-gitops-operator
+spec:
+  config:
+    env: 
+    - name: CONTROLLER_CLUSTER_ROLE 
+      value: gitops-controller-role <1>
```

Comment:
Callouts should be commented out. See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#using-asciidoc-callouts-in-the-yaml.
```suggestion
      value: gitops-controller-role # <1>
```